src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,46 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ARRAY_INDEX = 10;
+    private int size;
+    private final K[] keys;
+    private final V[] value;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_ARRAY_INDEX];
+        value = (V[]) new Object[MAX_ARRAY_INDEX];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (indexOf(key) != -1) {
```

Comment:
It's better to avoid calling method indexOf()twice
```suggestion
        int index = indexOf(key);
        if (index != -1) {
            values[index] = value;
        } else if (index == -1) {
            keys[size] = key;
            values[size] = value;
            size++;
        }
```